File: Digital Business Enablement/buy.it-api/src/main/java/br/com/fiap/buy/it/model/Historico.java
package br.com.fiap.buy.it.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;

import lombok.*;

import java.math.BigDecimal;
import java.util.Date;

@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@ToString
@Entity
@Table(name = "HISTORICO")
public class Historico {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_HISTORICO")
    @SequenceGenerator(name = "SQ_HISTORICO", sequenceName = "SQ_HISTORICO", allocationSize = 1)
    @Column(name = "ID_HISTORICO")
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinColumn(
            name = "ID_COTACAO",
            referencedColumnName = "ID_COTACAO",
            foreignKey = @ForeignKey(name = "FK_COTACAO_HISTORICO"),
            nullable = false
    )
    private Cotacao cotacao;

    @NotBlank(message = "O id do fornecedor não pode estar vazio.")
    @ManyToOne(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinColumn(
            name = "ID_FORNECEDOR",
            referencedColumnName = "ID_USUARIO",
            foreignKey = @ForeignKey(name = "FK_FORNECEDOR_HISTORICO"),
            nullable = false
    )
    private Usuario fornecedor;

    @NotBlank(message = "O id do status não pode estar vazio.")
    @ManyToOne(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinColumn(
            name = "ID_STATUS",
            referencedColumnName = "ID_STATUS",
            foreignKey = @ForeignKey(name = "FK_STATUS_HISTORICO"),
            nullable = false
    )
    private Status status;

    @Column(name = "RECUSADO_POR_PRODUTO", nullable = false)
    private Boolean recusadoPorProduto;

    @Column(name = "RECUSADO_POR_QUANTIDADE", nullable = false)
    private Boolean recusadoPorQuantidade;

    @Column(name = "RECUSADO_POR_PRECO", nullable = false)
    private Boolean recusadoPorPreco;

    @Column(name = "RECUSADO_POR_PRAZO", nullable = false)
    private Boolean recusadoPorPrazo;

    @Column(name = "DESCRICAO_HISTORICO", length = 400)
    private String descricao;

    @Column(name = "DATA_HISTORICO", nullable = false)
    @PastOrPresent
    private Date data;

    @Column(name = "VALOR_OFERTADO_HISTORICO")
    @Positive
    private BigDecimal valorOfertado;
}

File: Enterprise Application Development/buyit/Models/Historico.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace br.com.fiap.buy.it.model
{
    [Table("HISTORICO")]
    public class Historico
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("ID_HISTORICO")]
        public long Id { get; set; }

        [Required(ErrorMessage = "O campo cotacao não pode estar vazio.")]
        public Cotacao Cotacao { get; set; }

        [Required(ErrorMessage = "O campo fornecedor não pode estar vazio.")]
        public Usuario Fornecedor { get; set; }

        [Required(ErrorMessage = "O campo status não pode estar vazio.")]
        public Status Status { get; set; }

        [Required(ErrorMessage = "O campo recusadoPorProduto não pode estar vazio.")]
        [Column("RECUSADO_POR_PRODUTO")]
        public bool RecusadoPorProduto { get; set; }

        [Required(ErrorMessage = "O campo recusadoPorQuantidade não pode estar vazio.")]
        [Column("RECUSADO_POR_QUANTIDADE")]
        public bool RecusadoPorQuantidade { get; set; }

        [Required(ErrorMessage = "O campo recusadoPorPreco não pode estar vazio.")]
        [Column("RECUSADO_POR_PRECO")]
        public bool RecusadoPorPreco { get; set; }

        [Required(ErrorMessage = "O campo recusadoPorPrazo não pode estar vazio.")]
        [Column("RECUSADO_POR_PRAZO")]
        public bool RecusadoPorPrazo { get; set; }

        [MaxLength(400)]
        [Column("DESCRICAO_HISTORICO")]
        public string Descricao { get; set; }

        [Required(ErrorMessage = "O campo data não pode estar vazio.")]
        [Column("DATA_HISTORICO", TypeName = "date")]
        public DateTime Data { get; set; }

        [Column("VALOR_OFERTADO_HISTORICO")]
        public decimal? ValorOfertado { get; set; }
    }
}